### PR TITLE
Rename JobRunner to ForwardModelRunner

### LINKED ISSUE #7511: Rename _ert_job_runner
The "job-runner" should be renamed, as we no longer call the indivual steps in each realizations forward model for "Jobs". 

The job_runner happens on the compute node side, and the concept _Job_ is reserved for the whole thing that is running on the compute node on behalf of Ert.

https://github.com/equinor/ert/blob/39dbee19267a55a48c8f6d2cd24370fad80f11d3/src/_ert_job_runner/runner.py#L7

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #7511

**Approach**
We no longer call the individual steps in each realization forward model for `Jobs`, and the class name should be changed to reflect that. The commit in this PR fixes that.

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by berland** [comment]
I think this should be accompanied with a walk-through of *every* use of the string  "job" in this/these files, maybe only leaving "jobs.json" behind as it is breaking to change it. Do we want to do it in one PR or multiple?

File: src/_ert_job_runner/runner.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from _ert_job_runner.reporting.message import Finish, Init
 
 
-class JobRunner:
```
[the remark this answers is not in the document]
Comment (jonathan-eq) [reply src/_ert_job_runner/runner.py:7]: I don't think so. It is the actual runner, and it's in the _ert_forward_model_runner/ directory anyways.

**Comment by xjules** [comment]
> I think this should be accompanied with a walk-through of _every_ use of the string "job" in this/these files, maybe only leaving "jobs.json" behind as it is breaking to change it. Do we want to do it in one PR or multiple?

I agree to leave it to another PR covered by #7105

**Review by xjules — APPROVED** [review]
Good job @jonathan-eq ! :rocket:

**Pull request merged**